react-instantsearch-hooks/responsive-filtering/src/ResponsiveFilters/Filters.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useRef, useEffect } from 'react';
+import {
+  ToggleRefinement,
+  useInstantSearch,
+  HierarchicalMenu,
+  RefinementList,
+} from 'react-instantsearch-hooks-web';
+
+import { Panel } from './Panel';
+
+export function Filters() {
+  const { uiState, setUiState } = useInstantSearch();
+  const uiStateRef = useRef(uiState);
+
+  // Keep up-to-date `uiState` in a ref
+  useEffect(() => {
+    uiStateRef.current = uiState;
+  }, [uiState]);
+
+  // Apply latest `uiState` to InstantSearch as the component is unmounted
+  useEffect(() => {
+    return () => {
+      setTimeout(() => setUiState(uiStateRef.current));
+    };
+  }, [setUiState]);
+
+  return (
+    <>
+      <Panel header="category">
```

Comment:
Do we have the possibility to display each refinement in a collapsed way by default? If so, it's probably something we want to encourage by adding it by default, as filter panels get often crowed with many filters. Allowing end-users to more easily get an understanding of the filters available to them